File: src/lib/software/playnite.py
from src.lib.software import BaseSoftware

class Playnite(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'PlayniteInstaller.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://playnite.link/download/PlayniteInstaller.exe')

File: packages/app/src/services/software/definitions/playnite.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'playnite';
	public name = 'Playnite';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'PlayniteInstaller.exe';
	public icon = 'playnite.png';
	public homepage = 'https://playnite.link';

	public async resolveDownloadUrl() {
		return ok('https://playnite.link/download/PlayniteInstaller.exe');
	}
}
